public_html/wp-content/mu-plugins/blocks/assets/src/sponsors/edit.js

```diff
@@ -0,0 +1,90 @@
+/**
+ * Displays sponsor block.
+ */
+import SponsorInspectorControls from './inspector-controls';
+import SponsorBlockControls from './block-controls';
+import GridToolbar from '../shared/grid-layout/toolbar';
+
+/**
+ WordPress dependencies.
+ **/
+const { withSelect } = wp.data;
+const { Component, Fragment } = wp.element;
+const apiFetch = wp.apiFetch;
+const { addQueryArgs } = wp.url;
+
+const MAX_PAGE = 100;
+
+class SponsorsEdit extends Component {
+
+	/**
+	 * Constructor for SponsorsEdit block.
+	 *
+	 * @param props
+	 */
+	constructor( props ) {
+		super();
+
+		this.state = {};
+	}
+
+	componentWillMount() {
+		const sponsorQuery = {
+			orderby : 'title',
+			order   : 'asc',
+			per_page: MAX_PAGE,
+			_embed  : true,
+		};
+
+		const sponsorLevelQuery = {
+			orderby : 'id',
+			order: 'asc',
+			per_page: MAX_PAGE,
+			_embed: true
+		};
+
+		this.setState(
+			{
+				sponsorPosts: apiFetch( { path: addQueryArgs( '/wp/v2/sponsors', sponsorQuery ) } ),
+				sponsorLevels: apiFetch( { path: addQueryArgs( '/wp/v2/sponsor_level', sponsorLevelQuery ) } ),
+				siteSettings: apiFetch( { path: addQueryArgs( '/wp/v2/settings', {} ) } ),
+			}
```

Comment:
`[MIGHT]` @vedanshujain why is this done differently than the Speaker's block? I'm not implying it's wrong, I just want to understand the reason why it's different. If this is better, then should be do it in the other blocks as well, for consistency?

cc @coreymckrill

Reply:
+1 for consistency, one way or the other.

Reply:
So this is so that I can use results of same API call for rendering options in drop down, as well as rendering posts. Also, `withSelect` feels a bit opaque to me (it doesn't have good documentation as of now) so I wanted to avoid it if I can.

Reply:
Related: #39

Reply:
Possibly related: https://github.com/WordPress/gutenberg/pull/11300#issuecomment-437925427

Reply:
We have decided to use `withSelect` because it can cache request across blocks. Its being tracked in https://github.com/WordPress/wordcamp.org/issues/65 . Resolving this conversation.